homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +89,204 @@ def setup(hass, config):
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
 
-    zapi = ZabbixAPI(url)
+    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
+
+    include = conf.get(CONF_INCLUDE, {})
+    exclude = conf.get(CONF_EXCLUDE, {})
+    whitelist_e = set(include.get(CONF_ENTITIES, []))
+    whitelist_d = set(include.get(CONF_DOMAINS, []))
+    blacklist_e = set(exclude.get(CONF_ENTITIES, []))
+    blacklist_d = set(exclude.get(CONF_DOMAINS, []))
+
+    max_tries = conf.get(CONF_RETRY_COUNT)
+
     try:
-        zapi.login(username, password)
+        zapi = ZabbixAPI(url=url, user=username, password=password)
         _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
     except ZabbixAPIException as login_exception:
         _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
         return False
 
     hass.data[DOMAIN] = zapi
+
+    def event_to_metrics(event, float_keys, string_keys):
+        """Add an event to the outgoing Zabbix list."""
+        state = event.data.get("new_state")
+        entity_id = state.entity_id
+        if (
+            state is None
+            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE)
+            or entity_id in blacklist_e
+            or state.domain in blacklist_d
+        ):
+            return
+
+        floats = {}
+        strings = {}
+        try:
+            if (
+                (whitelist_e or whitelist_d)
+                and entity_id not in whitelist_e
+                and state.domain not in whitelist_d
+            ):
+                return
+
+            _state_as_value = float(state.state)
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            try:
+                _state_as_value = float(state_helper.state_as_number(state))
+                floats[entity_id] = _state_as_value
+            except ValueError:
+                strings[entity_id] = state.state
+
+        for key, value in state.attributes.items():
+            # For each value we try to cast it as float
+            # But if we can not do it we store the value
+            # as string
+            attribute_id = f"{entity_id}/{key}"
```

Comment:
I am not 100% certain but slash characters here might cause issues later.

Reply:
As far as I know entity_ids cannot contain slashes. An keys can't either. Right? If so this shouldn't be a problem. But we can use any other character here, too (except '.'). Which one is better suited?

Reply:
Maybe underscore '_' or hypen '-' is a lower risk separator but I don't know the full consequences of leaving the '/'.  It could be fine, in which case ignore me!

(I don't know homeassistant that well - just trying to help the other guys out by doing a bit of reviewing!)

Reply:
Underscores can by used in entity_id, that's not good. A hyphen '-' may be an option although I think it's very similar to '/' so I think it's fine to stick with it.